pkg/currency/module.go

```diff
@@ -27,16 +30,29 @@ func NewCurrencyModule() kernel.ModuleFactory {
 		module := &Module{
 			logger:         logger,
 			updaterService: updater,
+			healthy:        &atomic.Bool{},
 		}
 
 		return module, nil
 	}
 }
 
+func (module *Module) IsHealthy(ctx context.Context) (bool, error) {
+	return module.healthy.Load(), nil
+}
+
 func (module *Module) Run(ctx context.Context) error {
-	ticker := time.NewTicker(time.Duration(1) * time.Hour)
+	defer func() {
+		module.healthy.Store(false)
+	}()
+
+	// complete one refresh, then the module is healthy
```

Comment:
Hmm, but refresh and importExchangeRates don't ensure the operation is actually successful, so if they throw an error, the module becomes healthy anyway